src/asset/material/builder.rs

```diff
@@ -367,6 +367,66 @@ fn build_water_material(
     builder.build()
 }
 
+fn build_modulate_material(
+    context: &mut Context<BlenderAssetHandler>,
+    vmt: &VmtHelper,
+    settings: Settings,
+) -> BuiltMaterialData {
+    let shader = vmt.shader().shader.as_uncased_str();
+    let is_mod2x = shader == "decalmodulate".as_uncased()
+        || (shader == "modulate".as_uncased() && vmt.extract_param_or_default::<bool>("$mod2x"));
+
+    let mut builder = if is_mod2x {
+        // For mod2x materials, use transparent shader for proper blending
+        MaterialBuilder::new(&shaders::TRANSPARENT)
+    } else {
+        // For regular modulate, use principled shader with multiply blend
+        MaterialBuilder::new(&shaders::PRINCIPLED)
```

Comment:
Principled is probably not correct here. If mod2x uses transparent with 2x multiplication, maybe modulate should use transparent directly? Check this for verification:  https://developer.valvesoftware.com/wiki/Modulate